recipe/meta.yaml

```
@@ -41,25 +44,36 @@ build:
   skip: true  # [win]
   skip: true  # [py<35]
   detect_binary_files_with_prefix: true
+  {% if mpi != 'nompi' %}
+  {% set mpi_prefix = "mpi_" + mpi %}
+  {% else %}
+  {% set mpi_prefix = "nompi" %}
+  {% endif %}
 
   # add build string so packages can depend on
   # dagmc or nodagmc variants explicitly:
   # `openmc * nodagmc_*` for openmc without dagmc
   # `openmc * dagmc_*` for openmc with dagmc
+  # mpi or nompi variants explicitly:
+  # `openmc * mpi_mpich_*` for mpich
+  # `openmc * mpi_*` for any mpi
+  # `openmc * nompi_*` for no mpi
 
-  string: {{ dagmc }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build }}
+  string: {{ dagmc }}_{{ mpi_prefix }}_py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ build }}
 
 requirements:
   build:
     - cmake
     - make
-    - dagmc 3.2.1  # [dagmc == 'dagmc']
+    - dagmc * 3.2.1 * {{ mpi_prefix }}_*  # [dagmc == 'dagmc']
     - pkg-config
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - {{ mpi }}  # [mpi != 'nompi']
   host:
-    - dagmc 3.2.1  # [dagmc == 'dagmc']
+    - dagmc * 3.2.1 * {{ mpi_prefix }}_*  # [dagmc == 'dagmc']
```

Shouldn't this be:
```suggestion
    - dagmc 3.2.1 {{ mpi_prefix }}_*  # [dagmc == 'dagmc']
```
Looking at the documentation on [package match specifications](https://conda.io/projects/conda/en/latest/user-guide/concepts/pkg-specs.html#package-match-specifications), there should only be at most three things, so I'm not sure what the extra asterisks would do here.

---

I have no idea. It was all trial and error and this worked.